File: aop/src/main/java/com/example/aop/annotation/CrashSafe.java
package com.example.aop.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/***
 * 捕获Crash
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.CLASS)
public @interface CrashSafe {
}

File: aop/src/main/java/com/example/aop/annotation/CrashSafe.kt
package com.example.aop.annotation

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/***
 * 捕获Crash
 */
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER
)
@Retention(
    RetentionPolicy.CLASS
)
annotation class CrashSafe